File: Selenium/Activities/Activity15.java
/*Open a new browser to https://training-support.net/webelements/dynamic-attributes
Get the title of the page and print it to the console.
Find the input fields and type in the required data in the fields.
Wait for success message to appear and print it to the console.
Close the browser.*/
package training;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity15 {
	public static void main(String[] args) {
		
		//Initialize the web driver
		WebDriver driver = new FirefoxDriver();
		//Create the wait object
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		
		//Open the web page
		driver.get("https://training-support.net/webelements/dynamic-attributes");
		
		//Print the title of the page
		System.out.println("Title of the page: " + driver.getTitle());
		
		//Find the input fields
		WebElement fullName = driver.findElement(By.xpath("//input[starts-with(@id, full-name)]"));
		WebElement email = driver.findElement(By.xpath("//input[contains(@id, '-email')]"));
		WebElement eventDate = driver.findElement(By.xpath("//input[contains(@name, '-event-date-')]"));
		WebElement additionalDetails = driver.findElement(By.xpath("//textarea[contains(@id, '-additional-details-')]"));
		
		//Enter the details
		fullName.sendKeys("Nishanthi");
		email.sendKeys("nisha_fstcheck@gmail.com");
		eventDate.sendKeys("2025-01-08");
		additionalDetails.sendKeys("Testing");
		
		//Find and click the submit button
		WebElement submit = driver.findElement(By.xpath("//button[text() = 'Submit']"));
		submit.click();
		
		//Wait for success message and print it
		String message = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("action-confirmation"))).getText();
		System.out.println("Success message: " + message);
		
		//Close the browser
		driver.quit();
		
		
		
		
	}
}

File: Selenium/Activities/Activity15.py
"""Open a new browser to https://training-support.net/webelements/dynamic-attributes
Get the title of the page and print it to the console.
Find the input fields and type in the required data in the fields.
Wait for success message to appear and print it to the console.
Close the browser."""

#import webdriver
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

#Initialise the driver
with webdriver.Firefox() as driver:
    #Open the web page
    driver.get("https://training-support.net/webelements/dynamic-attributes")
    #Print the title of the page
    print("Title of the page: ", driver.title)
    
    #Find the input fields
    fullname = driver.find_element(By.XPATH, "//input[starts-with(@id, 'full-name')]")
    email = driver.find_element(By.XPATH, "//input[contains(@id, '-email')]")
    eventDate = driver.find_element(By.XPATH, "//input[contains(@name, '-event-date-')]")
    additionalDetails = driver.find_element(By.XPATH, "//textarea[contains(@id, '-additional-details-')]")

    #Enter the details
    fullname.send_keys("Nishanthi")
    email.send_keys("nishanthi.fstcheck@gmail.com")
    eventDate.send_keys("2025-01-08")
    additionalDetails.send_keys("Testing")

    #Find the submit button and click it
    driver.find_element(By.XPATH, "//button[text()='Submit']").click()

    #Wait for success message and print it
    message = wait.until(EC.visibility_of_element_located((By.ID, "action-confirmation"))).text
    print("Success message: ", message)
